src/uthash.h

```diff
@@ -999,6 +999,38 @@ for(((el)=(head)), ((tmp)=DECLTYPE(el)((head!=NULL)?(head)->hh.next:NULL));
   (el) != NULL; ((el)=(tmp)), ((tmp)=DECLTYPE(el)((tmp!=NULL)?(tmp)->hh.next:NULL)))
 #endif
 
+#ifdef NO_DECLTYPE
+#define HASH_TOP(hh,head,el,tmp,eof)                                             \
+do {                                                                             \
+  ((el)=(head));                                                                 \
+  ((*(char**)(&(tmp)))=(char*)((head!=NULL)?(head)->hh.next:NULL));              \
+  ((*eof)=((el) == NULL));                                                       \
+} while(0)
+#else
+#define HASH_TOP(hh,head,el,tmp,eof)                                             \
+do {                                                                             \
+  ((el)=(head));                                                                 \
+  ((tmp)=DECLTYPE(el)((head!=NULL)?(head)->hh.next:NULL));                       \
+  ((*eof)=((el) == NULL));                                                       \
+} while(0)
+#endif
+
+#ifdef NO_DECLTYPE
+#define HASH_NEXT(hh,head,el,tmp,eof)                                            \
+do {                                                                             \
+  ((el)=(tmp));                                                                  \
+  ((*(char**)(&(tmp)))=(char*)((tmp!=NULL)?(tmp)->hh.next:NULL));                \
+  ((*eof)=((el) == NULL));                                                       \
+} while(0)
+#else
+#define HASH_NEXT(hh,head,el,tmp,eof)                                            \
+do {                                                                             \
+  ((el)=(tmp));                                                                  \
+  ((tmp)=DECLTYPE(el)((tmp!=NULL)?(tmp)->hh.next:NULL));                         \
+  ((*eof)=((el) == NULL));                                                       \
+} while(0)
```

Comment:
Style nit: here and on line 1024, please add a space after the word `while` in `} while (0)`.